sequencer/src/state.rs

```diff
@@ -139,13 +140,25 @@ impl ValidatedState {
 }
 
 pub fn validate_and_apply_proposal(
+    expected_chain_config: ChainConfig,
     state: &mut ValidatedState,
     delta: &mut Delta,
     parent_leaf: &Leaf,
     proposal: &Header,
     receipts: Vec<FeeInfo>,
 ) -> anyhow::Result<()> {
     let parent_header = parent_leaf.get_block_header();
+
+    // validate `ChainConfig`
+    anyhow::ensure!(
+        proposal.chain_config == expected_chain_config.into(),
```

Comment:
Due to the way `PartialEq` works on `ResolvableChainConfig`, I think this will only pass if `proposal.chain_config` is exactly `expected_chain_config` (the full thing). I think this should be something like `proposal.chain_config.commit() == expected_chain_config.commit()`